src/dataflow/src/render/reduce.rs

```diff
@@ -523,6 +540,50 @@ where
         (key, values)
     });
 
+    if monotonic {
+        use differential_dataflow::operators::consolidate::ConsolidateStream;
+        use timely::dataflow::operators::Map;
+
+        let collection = collection
+            .consolidate()
+            .inner
+            .map(move |((key, values), time, diff)| {
+                assert!(diff > 0);
+                let aggr_funcs = aggr_funcs.clone();
```

Comment:
I don't think you want this clone here. It will be cloned for each record, where I think you probably want one instance for the closure that can be re-used.

Reply:
I think we talked about this a few days ago but this type of thing is definitely a minefield for me. thanks!

just want to sanity check:

is that also an issue here

```
443     let collection = collection.map(move |(key, row)| {
444         let aggrs = aggrs.clone();
...
```

Reply:
Looks like it!

Reply:
fixed in both cases